Fix FlxG.removeCamera does not remove the camera from FlxG.cameras

Dovyski opened the pull request:
I've tested it using this demo: https://github.com/phmongeau/SplitScreen

Fix:
https://github.com/FlixelCommunity/flixel/issues/1
https://github.com/AdamAtomic/flixel/issues/231

Dovyski committed:
Fix FlxG.removeCamera does not remove the camera from FlxG.cameras

Fix:
https://github.com/FlixelCommunity/flixel/issues/1
https://github.com/AdamAtomic/flixel/issues/231

moly commented:
I would recommend using `FlxG._game.contains` to check if the camera sprite is added. It's much neater than catching exceptions.

Dovyski committed:
Tweaks to make code more robust

FlxG.cameraRemove() now uses FlxG._game.contains() to check if the camera exists in the display list (better than catching exceptions).

FlxCamera.destroy() checks if the "screen" property was not already destroyed before it tries to nullify it. It prevents a crash when FlxG.removeCamera() tries to remove and destroy an already removed (and destroyed) camera.

Dovyski commented:
I agree. That `try...catch` block was already there, but its a good idea to replace it. I used the same commit to fix a crash when `FlxG.removeCamera()` is called with an already removed (and destroyed) camera.
